File: src/main/java/dev/jarand/authprotectedrequests/annotation/EnableProtectedRequests.java
package dev.jarand.authprotectedrequests.annotation;

import dev.jarand.authprotectedrequests.AuthApiClient;
import dev.jarand.authprotectedrequests.AuthApiClientConfig;
import dev.jarand.authprotectedrequests.TokenService;
import dev.jarand.authprotectedrequests.WebSecurityConfig;
import org.springframework.context.annotation.Import;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Import({
        WebSecurityConfig.class,
        TokenService.class,
        AuthApiClient.class,
        AuthApiClientConfig.class})
public @interface EnableProtectedRequests {

    ProtectRequest[] protectedRequests();

    OpenRequest[] openRequests();
}

File: src/main/kotlin/dev/jarand/authprotectedrequests/annotation/EnableProtectedRequests.kt
package dev.jarand.authprotectedrequests.annotation

import dev.jarand.authprotectedrequests.WebSecurityConfig
import dev.jarand.authprotectedrequests.authapi.AuthApiClientConfig
import dev.jarand.authprotectedrequests.authapi.AuthApiClientImpl
import dev.jarand.authprotectedrequests.authapi.AuthApiClientMock
import dev.jarand.authprotectedrequests.jws.JwsServiceImpl
import dev.jarand.authprotectedrequests.jws.JwsServiceMock
import org.springframework.context.annotation.Import

@kotlin.annotation.Target(AnnotationTarget.CLASS)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Import(value = [WebSecurityConfig::class, JwsServiceImpl::class, JwsServiceMock::class, AuthApiClientImpl::class, AuthApiClientMock::class, AuthApiClientConfig::class])
annotation class EnableProtectedRequests(val protectedRequests: Array<ProtectRequest>, val openRequests: Array<OpenRequest>)
